### PR TITLE
6 refactor get corevectors

### PR DESCRIPTION
**Pull request by lorenzoCapelli**

### CODE REVIEW

File: peepholelib/coreVectors/activations.py [hunk 1]
```diff
@@ -88,33 +125,34 @@ def get_activations(self, **kwargs):
...
         act_dl = DataLoader(act_td, batch_size=bs, collate_fn = lambda x: x, shuffle=False) 
         
         if verbose: print('Computing activations')
         
-        for cvs_data, act_data in tqdm(zip(cvs_dl, act_dl), disable=not verbose, total=len(cvs_dl)):
+        for act_data in tqdm(act_dl, disable=not verbose, total=len(act_dl)):
             with torch.no_grad():
-                y_predicted = model(cvs_data['image'].to(device))
+                y_predicted = model(act_data['image'].to(device))
+                act_data['output'] = y_predicted
```
Comment (lsrosa) [inline peepholelib/coreVectors/activations.py:148]: this line ` act_data['output'] = y_predicted` should be inside the `if not has_pred:`

File: peepholelib/coreVectors/dataset_depreacted.py
Comment (lsrosa) [inline peepholelib/coreVectors/dataset_depreacted.py:1]: you can remove a file that is not used anymore. We can always restore previous versions with git

File: peepholelib/coreVectors/get_coreVectors.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            n_samples = len(self._corevds[ds_key])
+            if verbose: print('loaded n_samples: ', n_samples)
+        else:
+            n_samples = len(self._actds[ds_key])
+            if verbose: print('loader n_samples: ', n_samples) 
+            self._corevds[ds_key] = PersistentTensorDict(filename=file_path, batch_size=[n_samples], mode='w')
+
+
+        if verbose: print(f'\n ---- Getting core vectors for {ds_key}\n')
+        print(self._n_samples[ds_key])
```
Comment (lsrosa) [inline peepholelib/coreVectors/get_coreVectors.py:56]: clean random prints

File: peepholelib/models/model_wrap.py [hunk 3]
```diff
@@ -66,6 +66,7 @@ def __init__(self, **kwargs):
         
         # set in set_model()
         self._model = None
+        self.num_classes = None
```
Comment (lsrosa) [inline peepholelib/models/model_wrap.py:69]: It seems like we are assuming that the model does not do the `softmax` and returns an `int` corresponding to a class. We gotta keep an eye on this.

**Review by lsrosa — CHANGES_REQUESTED** [review]
Just a couple of minor changes.

I will ask you to test the `get_in_activation()` as a `collate_fn` for the activations TD (see comments)

**Review by lsrosa — APPROVED** [review]
